website/src/common/MacroBenchmarkTable.tsx

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Badge } from "@/components/ui/badge";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { MacroDataValue, Range } from "@/types";
+import {
+  fixed,
+  formatByte,
+  formatGitRef,
+  secondToMicrosecond,
+} from "@/utils/Utils";
+import { Link } from "react-router-dom";
+
+type MacroBenchmarkTableDataRow = {
+  title: string;
+  old: MacroDataValue;
+  new: MacroDataValue;
+  p: number;
+  delta: number;
+  insignificant: boolean;
+};
+
+export type MacroBenchmarkTableData = {
+  qpsTotal: MacroBenchmarkTableDataRow;
+  qpsReads: MacroBenchmarkTableDataRow;
+  qpsWrites: MacroBenchmarkTableDataRow;
+  qpsOther: MacroBenchmarkTableDataRow;
+  tps: MacroBenchmarkTableDataRow;
+  latency: MacroBenchmarkTableDataRow;
+  errors: MacroBenchmarkTableDataRow;
+  totalComponentsCpuTime: MacroBenchmarkTableDataRow;
+  vtgateCpuTime: MacroBenchmarkTableDataRow;
+  vttabletCpuTime: MacroBenchmarkTableDataRow;
+  totalComponentsMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+  vtgateMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+  vttabletMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+};
+
+export type MacroBenchmarkTableProps = {
+  data: MacroBenchmarkTableData;
+  oldGitRef: string;
+  newGitRef: string;
+};
+
+const getDeltaBadgeVariant = (key: string, delta: number, p: number) => {
+  if (delta === 0) {
+    return "warning";
+  }
+  if (
+    key.includes("CpuTime") ||
+    key.includes("Mem") ||
+    key.includes("latency")
+  ) {
+    if (delta < 0) {
+      return "success";
+    }
+  }
+  if (key.includes("qps") || key.includes("tps")) {
+    if (delta > 0) {
+      return "success";
+    }
+  }
+  return "destructive";
+};
+
+const formatCellValue = (key: string, value: number) => {
+  if (key.includes("CpuTime")) {
+    return secondToMicrosecond(Number(fixed(value, 2)));
```

Comment:
The fixed call should be made on the output of `secondToMicrosecond` otherwise we remove most of the raw value. The raw value is usually for instance `0.0008856165`, keeping only 2 numbers after the comma will remove everything interesting. I fixed it via db030962